RegisterView.py

```diff
@@ -1,9 +1,80 @@
-from xml.etree.ElementTree import ElementTree
+import xml.etree.ElementTree as ElementTree
+import copy
 from os import path
 
+
 class RegisterView:
+  
+  def adapt_elements_to_peripheral(self, elements, new_peripheral_name, new_base_address):
+    new_elements = copy.deepcopy(elements) 
+    
+    for r in new_elements.findall('.//register'):
+      register_name = r.find('name').text
+      fullname = new_peripheral_name + '_' + register_name
+      r.set('fullname', fullname) 
+      # Calculate absolute address
+      offset = r.find('addressOffset').text 
+      r.set('address', hex(int(new_base_address,16) + int(offset,16)))
+   
+    return new_elements
+
+  def parse_svd_peripheral(self, peripheral, peripherals):
+    peripheral_name = peripheral.find('name').text
+    base_address = peripheral.find('baseAddress').text
+    try:
+      derived_from = peripheral.attrib['derivedFrom']
+      derive_matches = filter(lambda x: x.find('name').text == derived_from, peripherals)
+      if len(derive_matches) <= 0:
+        error_msg = "ERROR could not find peripheral %s info derived from %s" %(peripheral_name, derived_from)
+        raise Exception(error_msg)
+
+      # Create a duplicate in this peripheral
+      new_elements = self.adapt_elements_to_peripheral(derive_matches[0], peripheral_name, base_address)
+      peripheral.insert(0, new_elements)
+        
+      # No need to do anything else - complete copy has been made
+      return 
```

Comment:
yes, because all the stuff that happens below is irrelevant when the peripheral is derived from another one and a deep copy of it has been made :)